### PR TITLE
Add VertX BOM to Quarkus

### PR DESCRIPTION
**Pull request by cristianonicolai**
Adding vertx-web-graphql to bom so versions are aligned when using this library.

### CODE REVIEW

File: bom/runtime/pom.xml [hunk 1]
```diff
@@ -1665,6 +1665,11 @@
                 <artifactId>vertx-web-client</artifactId>
                 <version>${vertx.version}</version>
             </dependency>
+            <dependency>
+                <groupId>io.vertx</groupId>
+                <artifactId>vertx-web-graphql</artifactId>
+                <version>${vertx.version}</version>
```
Comment (gsmet) [inline bom/runtime/pom.xml]: @cescoffier @vietj I wonder if we should include the Vert.x bom in Quarkus?

Comment (cristianonicolai) [reply bom/runtime/pom.xml]: +1 very likely any other artifact that is used would need to be added anyway, using the bom definitely solve for more use cases.

Comment (cescoffier) [reply bom/runtime/pom.xml]: Yes, that would avoid many additional dependencies. 

However, we don't have a `bom` named `bom`. It's called `dep-chain`. The dependency to add is:

```
<dependency>
    <groupId>io.vertx</groupId>
    <artifactId>vertx-stack-depchain</artifactId>
    <version>3.8.0</version> <!-- must be replaced with the Vert.x version -->
    <type>pom</type>
</dependency>
```

Comment (gsmet) [reply bom/runtime/pom.xml]: @cristianonicolai I let you update your PR? We have a few boms already defined there: https://github.com/quarkusio/quarkus/blob/master/bom/runtime/pom.xml#L171

Comment (cristianonicolai) [reply bom/runtime/pom.xml]: sure, I can update my PR.

File: bom/runtime/pom.xml [hunk 2]
```diff
@@ -213,6 +213,15 @@
...
             </dependency>
 
+            <!-- Vert.X dependencies, imported as a BOM -->
+            <dependency>
+                <groupId>io.vertx</groupId>
+                <artifactId>vertx-stack-depchain</artifactId>
+                <version>${vertx.version}</version>
+                <scope>import</scope>
+                <type>pom</type>
+            </dependency>
```
Comment (gsmet) [inline bom/runtime/pom.xml]: @rsvoboda could you check that adding this BOM doesn't create dependency issues? Especially for Netty and Jackson as the Vert.x BOM also includes them.

Thanks!

Comment (rsvoboda) [reply bom/runtime/pom.xml]: will do ;) later today or tomorrow

**Review by rsvoboda — COMMENTED** [review]
I created few small projects, one depending on quarkus-jackson, second on quarkus-netty, third on quarkus-vertx and last one depending on all deps from https://github.com/vert-x3/vertx-stack/blob/3.7.1/stack-depchain/pom.xml.

Netty was resolved to version 4.1.34.Final in all relevant cases, Jackson databind resolved to 2.9.9.3.

vertx-stack-depchain BOM is added after netty-bom and jackson-bom in bom/runtime/pom.xml so I assume "correct" version should be resolved.

To check this idea I updated netty to  4.1.35.Final, built Quarkus and checked my sample apps. Updated version was present in all relevant dependency trees.

Looking at stack-depchain BOM makes me feel a bit uncertain as the **BOM defines 80 dependencies**.
 * Do we really want to import all of those 80 deps via Quarkus bom?
 * Will they work in native mode ?
 * What will be the support policy for these artifacts - e.g. CVE rating / fixes 

Other BOM imports like Jackson, RESTEasy, Netty, kubernetes-client-bom and JUnit are pulling a lot of stuff too, some of that stuff can be unwanted for Quarkus. Maybe a topic for wider discussion, especially if we think about 1.0 and post-1.0 era.

**Comment by cescoffier** [comment]
> Do we really want to import all of those 80 deps via Quarkus bom?

These are mainly the modules from the Vert.x ecosystem. 

> Will they work in native mode?

Not all of them, as for the other dependencies included in the BOM file. 
To be honest, I don't know which one will work OOTB in native mode (for sure config and discovery won't)

> What will be the support policy for these artifacts - e.g. CVE rating / fixes

The Vert.x team does a great job to update in a timely fashion.

**Comment by gsmet** [comment]
Well, in any case, we don't include the dependencies, we just ensure that they are all consistent.

Let's merge this, thanks @rsvoboda for checking the dependencies and thanks @cristianonicolai for the PR.

**Review by gsmet — APPROVED** [review]

**Pull request merged**